File: app/src/main/java/com/example/fabricfinder/Fabric.java
package com.example.fabricfinder;

import android.media.Image;

import java.util.List;

/**
 * @author Minka Firth
 */
public class Fabric {

    private final Image fabricPicture;
    private final int width;
    private final int height;
    private final int colour;
    private final List<String> tags;

    public Fabric(Image fabricPicture, int width, int height, int colour, List<String> tags) {
        this.fabricPicture = fabricPicture;
        this.width = width;
        this.height = height;
        this.colour = colour;
        this.tags = tags;
    }

    public Image getFabricPicture() {
        return fabricPicture;
    }

    public int getWidth() {
        return width;
    }

    public int getHeight() {
        return height;
    }

    public int getColour() {
        return colour;
    }

    public List<String> getTags() {
        return tags;
    }
}

File: app/src/main/java/com/example/fabricfinder/Fabric.kt
package com.example.fabricfinder

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * @author Minka Firth
 */
@Entity
data class Fabric(
        @PrimaryKey(autoGenerate = true)
        val id: Int = 0,
//        val image: String,
        val length: Int,
        val width: Int,
        val colour: String
)
